packages/Search/src/details/DTK_DetailsPriorityQueue.hpp

```diff
@@ -26,120 +29,94 @@ template <typename T>
-                break;
-            _heap[pos] = _heap[parent];
-            pos = parent;
-        }
-
-        _heap[pos] = elem;
-
-        // update the size of the heap
-        ++_size;
+        _heap.pushBack( value );
+        pushHeap( _heap.data(), _heap.data() + _heap.size(), _compare );
+    }
+    KOKKOS_INLINE_FUNCTION void push( ValueType &&value )
+    {
+        _heap.pushBack( std::move( value ) );
+        pushHeap( _heap.data(), _heap.data() + _heap.size(), _compare );
     }
-
     KOKKOS_INLINE_FUNCTION void pop()
     {
-        // ensure that the heap is not empty
-        assert( _size > 0 );
-
-        // replace the root with the last element on the last level
-        IndexType pos = 0;
-
-        // perform down-heap operation
-        while ( true )
-        {
-            // compare the new root with its children
-            // if they are in the correct order, stop
-            // if not, swap the element with one of its children and continue
-            IndexType const left_child = 2 * pos + 1;
-            IndexType const right_child = 2 * pos + 2;
-            IndexType next_pos = _size - 1;
-            for ( IndexType child : {left_child, right_child} )
-                if ( child < _size - 1 &&
-                     _compare( _heap[next_pos], _heap[child] ) )
-                    next_pos = child;
-            if ( next_pos == _size - 1 )
-                break;
-            _heap[pos] = _heap[next_pos];
-            pos = next_pos;
-        }
-
-        _heap[pos] = _heap[_size - 1];
-
-        // update the size of the heap
-        _size--;
+        popHeap( _heap.data(), _heap.data() + _heap.size(), _compare );
+        _heap.popBack();
     }
-
     // in TreeTraversal::nearestQuery, pop() is often followed by push which is
     // an opportunity for doing a single bubble-down operation instead of paying
     // for both one bubble-down and one bubble-up
     template <typename... Args>
-    KOKKOS_INLINE_FUNCTION void pop_push( Args &&... args )
+    KOKKOS_INLINE_FUNCTION void popPush( Args &&... args )
     {
-        assert( _size < _max_size );
-
-        // size will be decremented by pop()
-        _size++;
-
-        // put the new element at the bottom level
-        _heap[_size - 1] = T{std::forward<Args>( args )...};
-
-        // replace the root with that new element and bubble it down
-        pop();
+        assert( _heap.size() > 0 );
+        __bubbleDown( _heap.data(), std::ptrdiff_t( 0 ),
+                      std::ptrdiff_t( _heap.size() ),
+                      T{std::forward<Args>( args )...}, _compare );
     }
 
-    KOKKOS_INLINE_FUNCTION T const &top() const
+    // Accessors that shouldn't be there but that are convenient in
+    // TreeTraversal::nearestQuery()
+    KOKKOS_INLINE_FUNCTION typename Container::Pointer data()
+    {
+        return _heap.data();
+    }
+    KOKKOS_INLINE_FUNCTION ValueCompare const &value_comp() const
```

Comment:
Why `value_comp` does not follow the camel case style?

Reply:
I missed that one.